test/rules/experiments/nnbd/rules/null_check_on_nullable_type_parameter.dart

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N null_check_on_nullable_type_parameter`
+
+m1<T>(T p) => p!; // LINT
```

Comment:
I think these methods should have an explicit return type. As written, the return type is `dynamic`, which is potentially nullable, so the last part of the condition isn't actually being tested.